ledger/participant-integration-api/src/test/suite/scala/platform/store/backend/UpdateToDbDtoSpec.scala

```diff
@@ -0,0 +1,1191 @@
+        ec: ExecutionContext,
+        loggingContext: LoggingContext,
+    ): Future[CreatedEvent] = Future.failed(new RuntimeException("Not implemented"))
+    override def deserialize(raw: TreeEvent.Exercised, verbose: Boolean)(implicit
+        ec: ExecutionContext,
+        loggingContext: LoggingContext,
+    ): Future[ExercisedEvent] = Future.failed(new RuntimeException("Not implemented"))
+  }
+
+  // These test do not check the correctness of compression.
+  // All values are compressed using a dummy (identity) algorithm in this suite.
+  private[this] val compressionAlgorithmId = Some(123)
+  private[this] val compressionStrategy: CompressionStrategy = {
+    val noCompression = new FieldCompressionStrategy(compressionAlgorithmId, x => x)
+    CompressionStrategy(noCompression, noCompression, noCompression, noCompression)
+  }
+
+  private[this] val someParticipantId =
+    ParticipantId.assertFromString("UpdateToDbDtoSpecParticipant")
+  private[this] val otherParticipantId =
+    ParticipantId.assertFromString("UpdateToDbDtoSpecRemoteParticipant")
+  private[this] val someOffset = Offset.fromHexString(Ref.HexString.assertFromString("abcdef"))
+  private[this] val someRecordTime = Time.Timestamp.assertFromString("2000-01-01T00:00:00.000000Z")
+  private[this] val someApplicationId =
+    ApplicationId.assertFromString("UpdateToDbDtoSpecApplicationId")
+  private[this] val someCommandId = CommandId.assertFromString("UpdateToDbDtoSpecCommandId")
+  private[this] val someSubmissionId =
+    SubmissionId.assertFromString("UpdateToDbDtoSpecSubmissionId")
+  private[this] val someWorkflowId = WorkflowId.assertFromString("UpdateToDbDtoSpecWorkflowId")
+  private[this] val someConfiguration =
+    Configuration(1, TimeModel.reasonableDefault, Duration.ofHours(23))
+  private[this] val someParty = Party.assertFromString("UpdateToDbDtoSpecParty")
+  private[this] val someHash =
+    crypto.Hash.assertFromString("01cf85cfeb36d628ca2e6f583fa2331be029b6b28e877e1008fb3f862306c086")
+  private[this] val someArchive1 = DamlLf.Archive.newBuilder
+    .setHash("00001")
+    .setHashFunction(DamlLf.HashFunction.SHA256)
+    .setPayload(ByteString.copyFromUtf8("payload 1"))
+    .build
+  private[this] val someArchive2 = DamlLf.Archive.newBuilder
+    .setHash("00002")
+    .setHashFunction(DamlLf.HashFunction.SHA256)
+    .setPayload(ByteString.copyFromUtf8("payload 2 (longer than the other payload)"))
+    .build
+  private[this] val someSubmitterInfo = SubmitterInfo(
+    actAs = List(someParty),
+    someApplicationId,
+    someCommandId,
+    Instant.ofEpochMilli(1),
+  )
+  private[this] val someTransactionMeta = TransactionMeta(
+    ledgerEffectiveTime = Time.Timestamp.assertFromLong(2),
+    workflowId = Some(someWorkflowId),
+    submissionTime = Time.Timestamp.assertFromLong(3),
+    submissionSeed = someHash,
+    optUsedPackages = None,
+    optNodeSeeds = None,
+    optByKeyNodes = None,
+  )
```

Comment:
Minor, but if you think it would improve readability, consider putting helpers in a companion object.